design/codebases.go

```diff
@@ -77,6 +77,16 @@ var workspaceAttributes = a.Type("WorkspaceAttributes", func() {
 	a.Attribute("status", d.String, "The workspace status", func() {
 		a.Example("STOPPED");
 	})
+	a.Attribute("projects", a.ArrayOf(workpaceProject), "The workspace projects")
```

Comment:
Wouldn't a Che Project technically be a set of Codebases+a Branch?

Reply:
@aslakknutsen but codebases does not contain branch info.
Here is our discussion with @ashumilova regarding this:
```
- my start point is codebases - than i can fetch their workspaces. 
for example if i create workspace - with repo - eclipse/che and branch "test" - that full info will be only in `workspace.config.projects`
now when i fetch all workspaces per codebase - i do not get branch info, right? only workspaces info.
there is no place to retrieve the list of needed branches to request those two "get workspaces per branch" and "get workspaces per repo & banch"  - no input data about branch
there is way - to go to github api - fetch all branches for particular codebase - but there could be a lot of branches 
that are not needed for the user and no way to filter them

- ok, so the thing is that listWorkspaces should provide some project info

- as for me - yes , then will not need to get workspaces per branch get workspaces per repo & banch
```

+ there is still a possibility to create / import a project manually in che (not sure if it's considered to be supported usecase on osio), so for project retrieval per workspace it is still more reliable to use CHE API

Reply:
I understand the usecase, and I don't really care 'where' it's coming from.

Is 'Project' anything else than a Codebase+Branch?

Reply:
it is not, I believe. how branch info should be handled in this case ?
Also what about the manual import to IDE thingy - those projects should not be reflected in Codebases ?